gslib/utils/version_check.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions to ensure the correct version of Python is used."""
+
+import sys
+from sys import version_info as SYS_VER
+
+from gslib.utils import constants
+
+
+def check_python_version_support():
+  """Exit if gsutil is being run in an incompatible version of Python.
+
+  This function compares the running version of cPython and against the list
+  of supported python versions maintained in the constants file. If the running
+  version is less than any of the supported versions, exit.
+
+  Versions of Python greater than those listed in the currently supported
+  versions are implicitly allowed.
+  """
+  supported = constants.SUPPORTED_PYTHON_VERSIONS
```

Comment:
Discussed offline, the functionality in this module should be refactored a bit.

Reply:
Committed refactor!